encoders/firebase-encoders-json/src/main/java/com/google/firebase/encoders/FieldDescriptor.java

```diff
@@ -64,7 +64,12 @@ public String getName() {
   @Nullable
   @SuppressWarnings("unchecked")
   public <T extends Annotation> T getProperty(@NonNull Class<T> type) {
-    return (T) properties.get(type);
+    return type.cast(properties.get(type));
```

Comment:
curious why this change was needed?

Reply:
sorry, that was not intended. will remove it.

Reply:
looks like you forgot to remove it